lib/bech32/src/Codec/Binary/Bech32/Internal.hs

```diff
@@ -251,3 +345,17 @@ toBase256 :: [Word5] -> Maybe [Word8]
 toBase256 dat =
     map fromIntegral <$> convertBits (map fromWord5 dat) 5 8 noPadding
 
+{-------------------------------------------------------------------------------
+                                   Utilities
+-------------------------------------------------------------------------------}
+
+guardE :: Bool -> e -> Either e ()
+guardE b e = if b then Right () else Left e
```

Comment:
This is basically `when` from `Control.Monad`

```hs
when condition $
    return MyError
```

Reply:
> This is basically `when` from `Control.Monad`
> 
> ```haskell
> when condition $
>     return MyError
> ```

It's very close! Tantalizingly close. :)